tera-templates/crate/details.html

```diff
@@ -0,0 +1,303 @@
+                        {{ details.readme | safe }}
+
+                    {# If there's not a readme then attempt to display docs #}
+                    {%- elif details.rustdoc -%}
+                        {{ details.rustdoc | safe }}
+                    {%- endif -%}
+
+                {# If the build failed, the release isn't yanked and the release is a library #}
+                {%- else -%}
+                    {# Display a warning telling the user we failed to build the docs #}
+                    <div class="warning">
+                        docs.rs failed to build {{ details.name }}-{{ details.version }}<br>Please check the
+                            <a href="/crate/{{ details.name }}/{{ details.version }}/builds">build logs</a> and, if you believe this is
+                        docs.rs' fault, <a href="https://github.com/rust-lang/docs.rs/issues/new/choose">open an issue</a>.
+                    </div>
+
+                    {# If there is one, display the most next most recent successful build #}
+                    {%- if details.last_successful_build -%}
+                        <div class="info">
+                            Visit the last successful build:
+                            <a href="/crate/{{ details.name }}/{{ details.last_successful_build }}">
+                                    {{ details.name }}-{{ details.last_successful_build }}
+                            </a>
+                        </div>
+                    {%- endif -%}
+                {%- endif -%}
+            </div>
+        </div>
+    </div>
+{%- endblock body -%}
+
+{%- block css -%}
+    {{ macros::highlight_css() }}
+{%- endblock css -%}
+
+{%- block javascript -%}
+    {# Enable and load Rust and TOML syntax highlighting #}
+    {{ macros::highlight_js(languages=["rust", "ini"]) }}
+
+    <script type="text/javascript" charset="utf-8">
+        function getKey(ev) {
+            if ("key" in ev && typeof ev.key != "undefined") {
+                return ev.key;
+            }
+
+            return String.fromCharCode(ev.charCode || ev.keyCode);
+        }
+
+        document.getElementById("i-am-feeling-lucky-button").onclick = function () {
+            var form = document.getElementsByClassName("landing-search-form");
+            var input = document.createElement('input');
+
+            input.type = 'hidden';
+            input.name = 'i-am-feeling-lucky';
+            input.value = 1;
+            document.getElementsByClassName("landing-search-form")[0].appendChild(input);
+
+            return true;
+        };
+
+        function handleShortcut(ev) {
+            if (ev.ctrlKey || ev.altKey || ev.metaKey || document.activeElement.tagName === "INPUT") {
+                return;
+            }
+
+            switch (getKey(ev)) {
+                case "s":
+                case "S":
+                    ev.preventDefault();
+                    document.getElementById("search").focus();
+                    break;
+            }
```

Comment:
Was all this here already? This is a lot of JavaScript, it looks like rustdoc code.

Reply:
This was part of where crate pages and rustdoc pages melded together, I honestly couldn't say